io/mpi/inc/TMPIFile.h

```diff
@@ -0,0 +1,126 @@
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TMPIFile
+#define ROOT_TMPIFile
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TMPIFile                                                             //
+//                                                                      //
+// File Object derived from TMemFile.                                   //
+//                                                                      //
+// The TMPIFile class provides the ability to aggregate data across     //
+// many MPI ranks on a cluster into a single file. This can be useful   //
+// on HPCs or large clusters. The user must control the syncronization  //
+// of the data across multiple ranks via the Sync() function.           //
+// When Sync() is called, this triggers objects in the TFile space to   //
+// be communicated over MPI to a master writer which combines the data  //
+// before writing it to file.                                           //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#include "TMPIClientInfo.h"
+#include "TBits.h"
+#include "TFileMerger.h"
+#include "TMemFile.h"
+
+#include <mpi.h>
+
+#include <vector>
+
+class TMPIFile : public TMemFile {
+
+private:
+   Int_t fEndProcess = 0; // collector tracks number of exited processes
+   Int_t fSplitLevel;     // number of collectors to use
+   Int_t fMPIColor;       // used by MPI ranks to track which collector to use
+
+   Int_t fMPIGlobalRank; // global rank number
+   Int_t fMPIGlobalSize; // total ranks
+   Int_t fMPILocalRank;  // rank number in sub communicator
+   Int_t fMPILocalSize;  // number of ranks in sub communicator
+
+   MPI_Comm fSubComm;       // sub communicator handle
+   MPI_Request fMPIRequest; // request place holder
+
+   TString fMPIFilename; // output filename, only used by collector
+
+   char *fSendBuf = 0; // message buffer, only used by worker
+
+   struct ParallelFileMerger : public TObject {
+   private:
+      using ClientColl_t = std::vector<TMPIClientInfo>;
+
+      TString fFilename;
+      TBits fClientsContact;
+      UInt_t fNClientsContact;
+      ClientColl_t fClients;
+      TTimeStamp fLastMerge;
+      TFileMerger fMerger;
+      
+   public:
+      ParallelFileMerger(const char *filename, Int_t compression_settings, Bool_t writeCache = kFALSE);
+      virtual ~ParallelFileMerger();
+
+      ULong_t Hash() const { return fFilename.Hash(); };
+      const char *GetName() const { return fFilename; };
+
+      static void R__DeleteObject(TDirectory *dir, Bool_t withReset);
```

Comment:
Since these 2 functions are part of a class the naming convention is without the R_ prefix.  Also, do those 2 function need to be public?  Shoudn't they be private?

Reply:
OK, I rename the 2 functions you mentioned as well as R__MigrateKey() in TMPIClientInfo. DeleteObject is private now but NeedInitailMerge() remains public since it's called by the TMPIFile::RunCollector()